### PR TITLE
feat TDP-2966: implement 4 basic properties triggers

### PR DESCRIPTION
**Pull request by hmdebenque**
https://jira.talendforge.org/browse/TDP-2966

- [x] The commit message follows Talend standard
- [x] Tests for the changes have been added (for bug fixes / features)

**What is the new behavior?**

Now exists endpoint to access daikon properties triggers
- triggers implemented: validate, beforeActivate, beforeRender, after
- mapping: POST on /tcomp/{definitionName}/{callback}/{propName}

**Does this PR introduce a breaking change?**
- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by build-talend** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/886/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/886/)

File: services/components-api-service-rest/src/main/java/org/talend/components/service/rest/dto/CallbackType.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+ * 9 rue Pages 92150 Suresnes, France
+ */
+
+package org.talend.components.service.rest.dto;
+
+/**
+ * List possible callbacks on a property. As it is destined only to be serialized, in lower case, even if it would be preferable to have an
+ * enum serializer on jackson and/or spring to serialize in lower case.
+ */
+public enum CallbackType {
```
Comment (sgandon) [inline services/components-api-service-rest/src/main/java/org/talend/components/service/rest/dto/CallbackType.java]: Could you please math the constants used for UISchame serialization.
https://github.com/Talend/daikon/blob/master/daikon/src/main/java/org/talend/daikon/serialize/jsonschema/UiSchemaConstants.java#L35
Feel free to contribute to *daikon* if you prefer having an Enum.

**Comment by build-talend** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/894/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/894/)

File: services/components-api-service-rest/src/main/java/org/talend/components/service/rest/impl/PropertiesControllerImpl.java [hunk 2]
```diff
@@ -92,54 +103,68 @@ public String getProperties(@PathVariable("name") String definitionName) {
...
+                updatedProperties = componentService.beforePropertyActivate(propName, properties);
+                break;
+            case BEFORE_PRESENT:
+                updatedProperties = componentService.beforePropertyPresent(propName, properties);
+                break;
+            case AFTER:
+                updatedProperties = componentService.afterProperty(propName, properties);
+                break;
+            default:
+                throw new IllegalStateException("This enum does not contain this value: " + propertyTrigger);
```
Comment (sgandon) [inline services/components-api-service-rest/src/main/java/org/talend/components/service/rest/impl/PropertiesControllerImpl.java]: Could you throw a TalendRuntimeException here please ?

**Comment by build-talend** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/1012/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/1012/)

**Pull request merged**